sherpa/ui/tests/test_session.py

```diff
@@ -579,3 +581,317 @@ def test_error_estimate_not_run(name):
         func()
 
     assert str(exc.value) == "data set bob has not been set"
+
+
+def test_set_source_invalid():
+
+    s = Session()
+    with pytest.raises(ArgumentErr) as ae:
+        s.set_source('2 * made_up.foo')
+
+    assert str(ae.value) == "invalid model expression: name 'made_up' is not defined"
+
+
+def test_paramprompt_default():
+    """The default setting is False."""
+
+    s = Session()
+    assert not s._paramprompt
+
+
+@pytest.mark.parametrize("flag", [True, "yes", "yeS", 1, "1", "on", "ON"])
+def test_paramprompt_set_true(flag):
```

Comment:
Fine. I don think that @DougBurke is overdoing it here though. Do we really 7 tests for this? I know they will be fast, so I'm not asking to revert, but surely if "yeS" passes then  "yes" will pass, too.

Reply:
Note: It is of course possible to make "yeS" pass and "yes" fail, but I can't see any way that could happen accidentally and tests don't protect us against malicious acters who craft the most obscure code possible to introduce such subtle bugs into the code on purpose.

Reply:
So, the "boolean" check is added here in part for some potential future change when we might change the logic to be more-Pythonic (the existing logic comes from historical, including - but also pre-dating -S-Lang support)